File: apps/api/src/shared/events/order-placed.event.ts
export class OrderPlacedEvent {
  constructor(
    public readonly orderId: string,
    public readonly buyerId: string,
    public readonly total: string,
    public readonly couponCode: string | null,
    public readonly lines: Array<{
      productId: string;
      sellerId: string;
      quantity: number;
      unitPrice: string;
    }>,
  ) {}
}

File: src/main/java/com/minimalecommerce/shared/domain/event/OrderPlacedEvent.java
package com.minimalecommerce.shared.domain.event;

import java.math.BigDecimal;
import java.util.List;
import java.util.UUID;

public record OrderPlacedEvent(
        UUID orderId,
        UUID buyerId,
        String couponCode,
        BigDecimal total,
        List<Line> lines
) {
    public record Line(UUID productId, UUID sellerId, int quantity, BigDecimal unitPrice) {
    }
}
